midi/midi/encode.py

```diff
@@ -34,48 +48,140 @@ def get_events_from_array(track):
         add = [elem for elem in current_notes if elem not in notes]
 
         for elem in subtract:
-            events.append(Message('note_off', note=elem, velocity=VELOCITY, time=int(last_offset*TICKS_IN_BEAT)))
+            events.append(Message('note_off', note=elem, channel=track_channel,
+                                  velocity=VELOCITY, time=round(last_offset * acc)))
             last_offset = 0
         for elem in add:
-            events.append(Message('note_on', note=elem, velocity=VELOCITY, time=int(last_offset*TICKS_IN_BEAT)))
+            events.append(Message('note_on', note=elem, channel=track_channel,
+                                  velocity=VELOCITY, time=round(last_offset * acc)))
             last_offset = 0
 
         notes = current_notes
         last_offset += 1
 
     for elem in notes:
-        events.append(Message('note_off', note=elem, velocity=VELOCITY, time=int(last_offset*TICKS_IN_BEAT)))
+        events.append(Message('note_off', note=elem, channel=track_channel,
+                              velocity=VELOCITY, time=round(last_offset * acc)))
         last_offset = 0
 
-    events.append(MetaMessage('end_of_track', time=0))  # time=last_offset*TICKS_IN_BEAT
+    events.append(MetaMessage('end_of_track', time=0))
     return MidiTrack(events)
 
 
-def get_sequences(filepath):
-    return np.load(filepath, allow_pickle=True)
+def generate_file_from_2d_array(input_array, tempos, output_filename, accuracy):
+    if input_array.ndim != 2:
+        raise TypeError
```

Comment:
Is that an invariant or user error?

Reply:
this error is raised in case of incorrect user input - issue resolved in last commit (b511e4a4f5fde8eefa268bfec2d8c57854f2ec80)